src/Workspaces/Core/Portable/Utilities/EventMap.cs

```diff
@@ -160,10 +160,7 @@ internal EventHandlerSet(object registries)
                 _registries = (ImmutableArray<Registry<TEventHandler>>)registries;
             }
 
-            public bool HasHandlers
-            {
-                get { return _registries != null && _registries.Length > 0; }
```

Comment:
Should we be checking for `IsDefaultOrEmpty` here?  See https://github.com/dotnet/corefx/blob/master/src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs#L176, but it looks like `Length` throws on default instances.